auth/api/iam/session.go

```diff
@@ -30,11 +32,57 @@ type OAuthSession struct {
 	OwnDID                 did.DID
 	ClientState            string
 	RedirectURI            string
-	ServerState            map[string]interface{}
+	ServerState            ServerState
 	ResponseType           string
 	PresentationDefinition PresentationDefinition
 }
 
+// ServerState is a convenience type for extracting different types of data from the session.
+type ServerState map[string]interface{}
+
+const (
+	credentialMapStateKey = "credentialMap"
+	presentationsStateKey = "presentations"
+	submissionStateKey    = "presentationSubmission"
+)
+
+// VerifiablePresentations returns the verifiable presentations from the server state.
+// If the server state does not contain a verifiable presentation, an empty slice is returned.
+func (s ServerState) VerifiablePresentations() []vc.VerifiablePresentation {
+	presentations := make([]vc.VerifiablePresentation, 0)
+	if val, ok := s[presentationsStateKey]; ok {
+		// each entry should be castable to a VerifiablePresentation
+		if arr, ok := val.([]interface{}); ok {
+			for _, v := range arr {
+				if vp, ok := v.(vc.VerifiablePresentation); ok {
+					presentations = append(presentations, vp)
+				}
+			}
+		}
+	}
+	return presentations
+}
+
+// PresentationSubmission returns the Presentation Submission from the server state.
+func (s ServerState) PresentationSubmission() pe.PresentationSubmission {
+	if val, ok := s[submissionStateKey]; ok {
+		if pd, ok := val.(pe.PresentationSubmission); ok {
+			return pd
+		}
+	}
+	return pe.PresentationSubmission{}
+}
+
+// CredentialMap returns the credential map from the server state.
+func (s ServerState) CredentialMap() map[string]vc.VerifiableCredential {
+	if val, ok := s[credentialMapStateKey]; ok {
```

Comment:
If I remember correctly, you can shorten this to:

```go
if mapped, ok := s[credentialMapStateKey].(map[string]vc.VerifiableCredential); ok {
```

`nil` will return `!ok`